### PR TITLE
Link to mypy wiki page from CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by samestep**
Addresses one of the documentation points in #50513 by making it easier to find our `mypy` wiki page. Also updates the `CONTRIBUTING.md` table of contents and removes some trailing whitespace.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -291,6 +293,17 @@ pytest test/test_nn.py -k Loss -v
...
 
+### Running `mypy`
+
+One of the test suites runs `mypy` on the codebase:
+```bash
+python test/test_type_hints.py
+```
+See [Guide for adding type annotations to
+PyTorch](https://github.com/pytorch/pytorch/wiki/Guide-for-adding-type-annotations-to-PyTorch)
+for more information on how to set up `mypy` how to tackle type annotation
```
Comment (janeyx99) [inline CONTRIBUTING.md:304]:
```suggestion
for more information on how to set up `mypy`, how to tackle type annotation
```
added comma

**Review by janeyx99 — APPROVED** [review]
cool

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -291,6 +293,17 @@ pytest test/test_nn.py -k Loss -v
...
 
+### Running `mypy`
+
+One of the test suites runs `mypy` on the codebase:
+```bash
+python test/test_type_hints.py
+```
+See [Guide for adding type annotations to
+PyTorch](https://github.com/pytorch/pytorch/wiki/Guide-for-adding-type-annotations-to-PyTorch)
+for more information on how to set up `mypy`, how to tackle type annotation
```
Comment (samestep) [inline CONTRIBUTING.md:304]:
```suggestion
for more information on how to set up `mypy` and tackle type annotation
```

**Review by rgommers — APPROVED** [review]
Thanks @samestep, LGTM.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,10 +2,12 @@
 
 - [Contributing to PyTorch](#contributing-to-pytorch)
 - [Developing PyTorch](#developing-pytorch)
-  - [Nightly Checkout & Pull](#nightly-checkout--pull)
+  - [Tips and Debugging](#tips-and-debugging)
```
Comment (walterddr) [inline CONTRIBUTING.md:5]: wondering why this is not auto-generated :-(

Comment (samestep) [reply CONTRIBUTING.md:5]: the only reason I noticed it was out of sync is because VS Code auto-generated it while I was writing this PR ;)

**Pull request closed without merging**